### ISSUE TITLE
Cascaded table data is dropped on column add during migration

### ISSUE THREAD

**Issue by dominic-simplan** [typeorm/typeorm#2584]
**Issue type:**

[x] question
[x] bug report
[ ] feature request
[ ] documentation issue

**Database system/driver:**

[x] `cordova`
[ ] `mongodb`
[ ] `mssql`
[ ] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`
[ ] `expo`

**TypeORM version:**

[x] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

I have a tableA which has ON CASCADE DELETE foreign key contstraint on tableB.
Now I add a column to tableB during a migration, and all data of tableA is gone, as the migration drops and recreates tableB:

```
await queryRunner.addColumn("Project", new TableColumn({
            name: "CustomVersion",
            type: "varchar",
            isNullable: true
        }));

query:  CREATE TABLE "temporary_Project" (...)
query:  INSERT INTO "temporary_Project"...
query:  DROP TABLE "Project"
```

Shouldn't the foreign key constraints be handled/disabled before dropping the table? Or am I doing it wrong?

**Comment by dominic-simplan** [comment]
Also seems like `await queryRunner.query("PRAGMA foreign_keys=OFF;");` as a workaround has no effect as  the [foreign_key is always set to on](https://github.com/typeorm/typeorm/blob/47e8f875c3500614d7a93fdb05164b2939034d9f/src/driver/cordova/CordovaDriver.ts#L85) when creating a connection (and a new connection is created on every `queryRunner.query`?)

**Comment by dominic-simplan** [comment]
The workaround is to change the foreign_key before starting the migration:
```
await connection.query("PRAGMA foreign_keys=OFF;");
await connection.runMigrations();
await connection.query("PRAGMA foreign_keys=ON;");
```

The foreign_key cannot be changed in the migration itself as migrations are executed in a transaction and foreign_key cannot be changed within a transaction:

> It is not possible to enable or disable foreign key constraints in the middle of a multi-statement transaction (when SQLite is not in autocommit mode). Attempting to do so does not return an error; it simply has no effect. 
Source: https://www.sqlite.org/foreignkeys.html

**Comment by pleerock** [comment]
Interesting issue...

I'm not completely sure but we might try to disable foreign keys before table recreation [here in recreateTable method](https://github.com/pleerock/typeorm/blob/38b6e7a598b8c773da58e1e2c0be59ee7c613f2b/src/driver/sqlite-abstract/AbstractSqliteQueryRunner.ts#L959). Can you add it on your own and test everything, and if it works create a PR?

**Comment by dominicdesu** [comment]
@pleerock As far as I can see, the recreateTable method is executed within a transaction? Then it is not possible to change the foreign_key pragma, that needs to be done before the transaction is created...

**Comment by sinopsysHK** [comment]
Hello,

i apologise if I got confused: this issue seems to be raised for cordova but I do have the exact same issue with sqlite. No one else is having problems with this feature ?

**Comment by sinopsysHK** [comment]
Should be closed with #9775
